examples/aks/aks_cluster_azure_cni/variables.tf

```diff
@@ -0,0 +1,52 @@
+# AKS  cluster variables.
+variable "cluster_name" {
+  type        = string
+  description = "Name of the AKS cluster, resources will be created for."
+}
+
+variable "cluster_region" {
+  type        = string
+  description = "Region of the AKS cluster, resources will be created for."
+}
+
+variable "cluster_version" {
+  type        = string
+  description = "AKS cluster version."
+  default     = "1.23"
+}
+
+variable "cluster_network_provider" {
+  type        = string
+  description = "AKS cluster network provider"
+  default     = "azure"
+}
+
+variable "cluster_network_provider_mode" {
+  type        = string
+  description = "AKS cluster network provider mode"
+  default     = ""
```

Comment:
Shouldn't it be `Overlay` as a default given the example for azure cni?

Reply:
CNI has 2 modes. "" and Overlay. So by default its "", but you can do "Overlay" if you chose to. Considering its new feature I wanted to keep it not default now.